### PR TITLE
Sidebar context

### PR DESCRIPTION
**Pull request by msecret**
Doesn't fully implement 382 but is finished enough for a stopping point.

Ref #382

### CODE REVIEW

File: static_src/app.jsx [hunk 1]
```diff
@@ -53,7 +53,10 @@ export default class App extends React.Component {
 
     if (this.state.isLoggedIn) {
       content = this.props.children;
-      sidebar = <Nav initialCurrentOrgGuid={ this.props.currentOrgGuid } />;
+      sidebar = <Nav
+        initialCurrentOrgGuid={ this.props.currentOrgGuid }
+        initialSpaceGuid={ this.props.initialSpaceGuid }
```
Comment (jcscottiii) [inline static_src/app.jsx]: to be consistent with the naming, should this be `initialSpaceGuid={ this.props.currentSpaceGuid }`

File: static_src/components/navbar.jsx [hunk 2]
```diff
@@ -23,23 +27,33 @@ export class Nav extends React.Component {
...
     OrgStore.removeChangeListener(this._onChange);
+    SpaceStore.removeChangeListener(this._onChange);
   }
 
   _onChange() {
     this.setState(stateSetter());
   }
 
+  _handleOverviewClick() {
+    // orgActions.changeCurrentOrg('0');
```
Comment (jcscottiii) [inline static_src/components/navbar.jsx]: remove comment

File: static_src/components/navbar.jsx [hunk 3]
```diff
@@ -23,23 +27,33 @@ export class Nav extends React.Component {
...
+    orgActions.toggleSpaceMenu('0');
```
Comment (jcscottiii) [inline static_src/components/navbar.jsx]: could this be `toggleSpaceMenu` or change the following line to `toggleCurrentSpace`

File: static_src/stores/org_store.js [hunk 4]
```diff
@@ -73,9 +73,24 @@ class OrgStore extends BaseStore {
 
       case orgActionTypes.ORG_TOGGLE_SPACE_MENU: {
         const org = this.get(action.orgGuid);
+        if (!org) {
+          const allOrgs = this.getAll();
+          const updates = allOrgs.map((sorg) =>
```
Comment (jcscottiii) [inline static_src/stores/org_store.js]: `sorg` variable name is confusing

**Comment by jcscottiii** [comment]
just a few comments! @msecret

**Comment by jeremiak** [comment]
It doesn't seem right to me that the "Marketplace" would be nested at a level equal to the individual spaces since it applies to the whole organization.

Is there a corresponding cg-style PR/release? As it is now, the contrast isn't sufficient in the menu text and background color.

screen shot 2016-07-06 at 10 05 36 pm

**Comment by msecret** [comment]
@thisisdano is there a reason you changed the Marketplace link to be at the space level?

**Comment by thisisdano** [comment]
@msecret Hm yeah — that looks like a mistake. I was trying to match the comp and took the wrong cue from it.

**Comment by msecret** [comment]
Cool, I'll fix the merge conflicts and that and this will be ready to go

**Comment by jeremiak** [comment]
This looks pretty great! Can be merged when tests pass

**Pull request merged**